docs/design-documents/20241024-conduit-cli.md

```diff
@@ -0,0 +1,375 @@
+|------|-------------|----------|---------------|
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit pipelines ls
+NAME            STATUS
+my-pipeline     running
+my-other        degraded      
+```
+
+### `conduit pipelines describe ID`
+
+This command will describe the topology of the pipeline.
+
+It requires having Conduit previously running.
+
+It requires the pipeline id as argument.
+
+#### Arguments
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| id  |  pipeline id to describe | Yes | |
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit pipelines describe [ID]
+
+EXAMPLE:
+
+$ conduit pipelines describe my-pipeline
+Source: kafka
+  Processor: avro.encode
+Destination: kafka
+```
+
+### `conduit connectors ls`
+
+This command will list all the available connectors.
+
+It requires having Conduit previously running.
+
+#### `--help`
+
+```bash
+$ conduit connectors ls
+ID                PLUGIN                          TYPE         PIPELINE
+my-source         postgres@v0.2.0                  builtin      file-to-postgres
+my-destination    conduit-connector-http@0.1.0.   standalone   my-other-pipeline
+```
+
+### `conduit connectors describe [--plugin] [--id]`
+
+This command will describe the connector configuration available.
+
+When using `--plugin` will describe the configuration for that connector plugin.
+
+When using `--id` will require having Conduit previously running since it'll describe the existing running connector.
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| id  |  connector identifier (returned by `connectors ls`) | Yes | |
+| plugin  |  plugin name and version | Yes | |
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit connnectors describe [--plugin] [--id] [--grpc.address]
+
+EXAMPLE:
+
+$ conduit connectors describe --plugin conduit-connector-http@0.1.0
+NAME   DESCRIPTION                       REQUIRED  DEFAULT VALUE  EXAMPLE
+url    HTTP URL to send requests to.     true                     https://...
+...
+
+$ conduit connectors describe --id my-source
+NAME        URL             PIPELINE
+my-source   https://...     my-pipeline
+```
+
+### `conduit processors ls`
+
+This command will list all the available processors.
+
+It requires having Conduit previously running.
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit processors ls
+NAME          TYPE      PIPELINE
+avro.decode   builtin   my-pipeline
+avro.encode    builtin   my-pipeline
+base64.decode  builtin   my-other-pipeline
+```
+
+### `conduit processor describe [NAME]`
```

Comment:
Maybe we can make it work the same as `conduit connectors describe`, to have the `--plugin` and `--id`.